Buildings/Fluid/HeatExchangers/DXCoils/AirSource/Validation/SingleSpeedHeating_OnDemandResistiveDefrost.mo

```diff
@@ -0,0 +1,296 @@
+
+  parameter Modelica.Units.SI.Power Q_flow_nominal=datCoi.sta[1].nomVal.Q_flow_nominal
+    "Nominal power";
+
+  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal=datCoi.sta[1].nomVal.m_flow_nominal
+    "Nominal mass flow rate";
+
+  parameter Modelica.Units.SI.PressureDifference dp_nominal=1141
+    "Pressure drop at m_flow_nominal";
+
+  parameter Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.CoilHeatTransfer
+    datCoi(
+    is_CooCoi=false,
+      sta={Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.Stage(
+        spe=1800/60,
+        nomVal=
+          Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.NominalValues(
+          is_CooCoi=false,
+          Q_flow_nominal=15000,
+          COP_nominal=2.75,
+          SHR_nominal=1,
+          m_flow_nominal=0.782220983308365,
+          TEvaIn_nominal=273.15 + 6,
+          TConIn_nominal=273.15 + 21),
+        perCur=
+          Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Examples.PerformanceCurves.DXHeating_Curve_II())},
+      nSta=1)
+    "Coil heat transfer data"
+    annotation (Placement(transformation(extent={{80,40},{100,60}})));
+
+  Buildings.Fluid.Sources.Boundary_pT sin(
+    redeclare package Medium = Medium,
+    final p(displayUnit="Pa") = 101325,
+    final nPorts=1,
+    final T=294.15)
+    "Sink"
+    annotation (Placement(transformation(extent={{40,-20},{20,0}})));
+
+  Buildings.Fluid.HeatExchangers.DXCoils.AirSource.SingleSpeedHeating sinSpeDX(
+    redeclare package Medium = Medium,
+    final dp_nominal=dp_nominal,
+    datCoi(
+      final nSta=datCoi.nSta,
+      final minSpeRat=datCoi.minSpeRat,
+      final sta=datCoi.sta),
+    final T_start=datCoi.sta[1].nomVal.TEvaIn_nominal,
+    final from_dp=true,
+    final energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
+    final datDef=datDef)
+    "Single speed DX heating coil"
+    annotation (Placement(transformation(extent={{-10,0},{10,20}})));
+
+  Buildings.Utilities.IO.BCVTB.From_degC TEvaIn_K
+    "Converts degC to K"
+    annotation (Placement(transformation(extent={{-100,70},{-80,90}})));
+
+  Buildings.Utilities.IO.BCVTB.From_degC TConIn_K
+    "Converts degC to K"
+    annotation (Placement(transformation(extent={{-100,-20},{-80,0}})));
+
+  Modelica.Blocks.Math.Mean TOutMea(
+    final f=1/3600)
+    "Mean of measured outlet air temperature"
+    annotation (Placement(transformation(extent={{80,80},{100,100}})));
+
+  Buildings.Utilities.IO.BCVTB.To_degC TOutDegC
+    "Convert measured outlet air temperature to deg C"
+    annotation (Placement(transformation(extent={{120,80},{140,100}})));
+
+  Modelica.Blocks.Sources.RealExpression TOut(
+    final y=sinSpeDX.vol.T)
+    "Measured temperature of outlet air"
+    annotation (Placement(transformation(extent={{40,80},{60,100}})));
+
+  Modelica.Blocks.Math.Mean XConOutMea(
+    final f=1/3600)
+    "Mean of measured outlet air humidity ratio (Total air)"
```

Comment:
Use lower case, and better just write "... ratio per kg total air" as brackets stop the flow.